posydon/popsyn/binarypopulation.py

```diff
@@ -644,11 +644,15 @@ def from_hdf(self, indices, where=None, restore=False):
         where : str
             Query performed on disk to select history and oneline DataFrames.
         restore : bool
-            Restore binaries back to initial conditions.
+            If true, restore binaries back to initial conditions.
 
         """
         if where is None:
-            query_str = 'index==indices'
+            if len(indices) > 0:
+                query_str = f'index=={indices}'
```

Comment:
The original code had `query_str = 'index==indices'` but the edits have replaced this with an f string. Is there a reason this behavior was changed? In principle these query strings are executed as python code during the search in the h5 and can reference local variables.

Reply:
The f-string makes this behavior more readable.

Reply:
I think the original is actually more clear. If `indices` is an integer this code will fail. Using f-strings requires the conversion of `indices` to the string representation of the object before the logical evaluation which may raise errors down the road (e.g. interpreted as a float / int / string is unclear). In the original, the logical evaluation is executed without this step.

Reply:
I think this function should support integers (as it did with the original code). You also do not explicitly check that this will be a list of integers so any error raised will be opaque. You also have not provided any sufficient reasoning for why the string conversion may not cause issues downstream. These edge cases will need to be handled and checked that they do not impact IO performance.

Furthermore, the original code is what is described in the pandas documentation and this change would remove functionality for advanced queries within tables.

Reply:
You did not see my most recent commit, I changed things back.

Reply:
Going forward I would prefer discussion be handled in a tone that is less aggressive, thanks!

Reply:
Apologies!